src/Features/CSharp/Portable/CodeRefactorings/ConvertIfToSwitch/CSharpConvertIfToSwitchCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,246 @@
+                ExpressionSyntax syntaxNode)
+            {
+                syntaxNode = syntaxNode.WalkDownParentheses();
+                while (syntaxNode.IsKind(SyntaxKind.LogicalOrExpression))
+                {
+                    var binaryExpression = (BinaryExpressionSyntax)syntaxNode;
+                    yield return binaryExpression.Right.WalkDownParentheses();
+                    syntaxNode = binaryExpression.Left.WalkDownParentheses();
+                }
+
+                yield return syntaxNode;
+            }
+
+            protected override IEnumerable<(StatementSyntax, ExpressionSyntax)> GetIfElseStatementChain(IfStatementSyntax currentStatement)
+            {
+                StatementSyntax elseBody;
+                do
+                {
+                    yield return (currentStatement.Statement, currentStatement.Condition);
+                    elseBody = currentStatement.Else?.Statement;
+
+                    var elseIfStatement = elseBody as IfStatementSyntax;
+                    if (elseIfStatement == null
+                        && !_semanticModel.AnalyzeControlFlow(currentStatement.Statement).EndPointIsReachable
+                        && currentStatement.GetNextStatement() is IfStatementSyntax nextStatement
+                        && CanConvertIfToSwitch(nextStatement))
+                    {
+                        _numberOfSubsequentIfStatementsToRemove++;
+                        currentStatement = nextStatement;
+                    }
+                    else
+                    {
+                        currentStatement = elseIfStatement;
+                    }
+                }
+                while (currentStatement != null);
+
+                yield return (elseBody, null);
+            }
+
+            protected override IEnumerable<SyntaxNode> GetSubsequentIfStatements(IfStatementSyntax ifStatement)
+            {
+                StatementSyntax currentStatement = ifStatement;
+                for (int i = 0; i < _numberOfSubsequentIfStatementsToRemove; ++i)
+                {
+                    yield return currentStatement = currentStatement.GetNextStatement();
+                }
+            }
+
+            private static ExpressionSyntax GetLeftmostCondition(ExpressionSyntax syntaxNode)
+            {
+                syntaxNode = syntaxNode.WalkDownParentheses();
+                while (syntaxNode.IsKind(SyntaxKind.LogicalAndExpression))
+                {
+                    syntaxNode = ((BinaryExpressionSyntax)syntaxNode).Left.WalkDownParentheses();
+                }
+
+                return syntaxNode;
+            }
+
+            protected override IEnumerable<SyntaxNode> GetSwitchSectionBody(StatementSyntax node)
+            {
+                bool RequiresBreak() => _semanticModel.AnalyzeControlFlow(node).EndPointIsReachable;
+                bool RequiresBlock() => !_semanticModel.AnalyzeDataFlow(node).VariablesDeclared.IsDefaultOrEmpty;
```

Comment:
i don't see a lot of value in making these functions/lambdas.  Why not just compute and store the values up front?